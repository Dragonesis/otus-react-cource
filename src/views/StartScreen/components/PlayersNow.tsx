import React from 'react'
import styled from '@emotion/styled'

export interface PlayersNowProps {
  countGamer: number
}
export interface PlayersNowState {}

export class PlayersNow extends React.Component<PlayersNowProps, PlayersNowState> {
  constructor(props: PlayersNowProps) {
    super(props)

    this.state = {
      countGamer: 0,
    } as PlayersNowState
  }

  shouldComponentUpdate = (nextProps: PlayersNowProps): boolean => {
    return nextProps.countGamer !== this.props.countGamer
  }

  render() {
    const { countGamer } = this.props
    return <Core>Игроков в онлайне: {countGamer}</Core>
  }
}

const Core = styled.p`
  font-size: 22px;
  margin-top: 16px;
`
