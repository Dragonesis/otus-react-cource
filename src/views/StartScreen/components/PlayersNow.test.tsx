import React from 'react'
import { render, screen } from '@testing-library/react'

import { PlayersNow } from './PlayersNow'

describe('### PlayersNow', () => {
  test('PlayersNow renders correctly', () => {
    const { asFragment } = render(<PlayersNow countGamer={200} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('PlayersNow displays the declared data', () => {
    render(<PlayersNow countGamer={200} />)
    expect(screen.getByText('Игроков в онлайне: 200')).toBeInTheDocument()
    render(<PlayersNow countGamer={300} />)
    expect(screen.getByText('Игроков в онлайне: 300')).toBeInTheDocument()
  })

  test('PlayersNow rerender', () => {
    const { rerender } = render(<PlayersNow countGamer={200} />)
    rerender(<PlayersNow countGamer={200} />)
  })
})
