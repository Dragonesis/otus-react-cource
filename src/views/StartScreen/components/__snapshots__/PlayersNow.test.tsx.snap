// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`### PlayersNow PlayersNow renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 22px;
  margin-top: 16px;
}

<p
    class="emotion-0"
  >
    Игроков в онлайне: 200
  </p>
</DocumentFragment>
`;
